src/librustdoc/passes/lint/html_tags.rs

```diff
@@ -400,8 +412,147 @@ fn extract_tags(
+                // We don't have any code for parsing fully-qualified trait paths.
+                // In theory, we could add it, but doing it correctly would require
+                // parsing the entire path grammar, which is problematic because of
+                // overlap between the path grammar and Markdown.
+                //
+                // The example above shows that ambiguity. Is `[u32]` intended to be an
+                // intra-doc link to the u32 primitive, or is it intended to be a slice?
+                //
+                // If the below conditional were removed, we would suggest this, which is
+                // not what the user probably wants.
+                //
+                //     <[u32] as `IntoIter<u32>`>::Item
+                //
+                // We know that the user actually wants to wrap the whole thing in a code
+                // block, but the only reason we know that is because `u32` does not, in
+                // fact, implement IntoIter. If the example looks like this:
+                //
+                //     <[Vec<i32>] as IntoIter<i32>::Item
+                //
+                // The ideal fix would be significantly different.
+                if (generics_start > 0 && dox.as_bytes()[generics_start - 1] == b'<')
+                    || (generics_end < dox.len() && dox.as_bytes()[generics_end] == b'>')
+                {
+                    return lint;
+                }
+                // multipart form is chosen here because ``Vec<i32>`` would be confusing.
+                lint.multipart_suggestion(
+                    "try marking as source code",
+                    vec![
+                        (generics_sp.shrink_to_lo(), String::from("`")),
+                        (generics_sp.shrink_to_hi(), String::from("`")),
+                    ],
+                    Applicability::MaybeIncorrect,
+                );
+            }
+
+            lint
+        });
+    }
+    fn crate_name(&self) -> rustc_span::Symbol {
+        self.cx.tcx.crate_name(self.item.item_id.krate())
+    }
+    fn unprefixed_html_class(&self, range: &Range<usize>) {
+        let ReportDiag { cx, ref dox, hir_id, item } = *self;
+        let tcx = cx.tcx;
+        let span = match source_span_for_markdown_range(tcx, &dox, range, &item.attrs) {
+            Some(span) => span,
+            None => item.attr_span(tcx),
+        };
+        let msg = "unprefixed HTML `class` attribute";
+        tcx.struct_span_lint_hir(crate::lint::UNPREFIXED_HTML_CLASS, hir_id, span, msg, |lint| {
+            use rustc_lint_defs::Applicability;
+            lint.multipart_suggestion(
+                "add prefix",
+                vec![
+                    (span.shrink_to_lo(), format!("{}_", self.crate_name())),
+                ],
+                Applicability::MaybeIncorrect,
+            );
+            lint.help("classes should start with `{cratename}_`, or be: `stab`, `stab deprecated`, or `stab portability`");
```

Comment:
Wouldn't it be better to put the allowed class names into an array so it's easier to maintain? After all we're currently working on adding `warning` class available for users. ;)